src/detect-engine-tag.c

```diff
@@ -1171,54 +1100,38 @@ static int DetectTagTestPacket05 (void)
         TimeGet(&p[i]->ts);
         SigMatchSignatures(&th_v, de_ctx, det_ctx, p[i]);
 
-        if (UTHCheckPacketMatchResults(p[i], sid, (uint32_t *)&results[i][0], numsigs) == 0)
-            goto cleanup;
+        FAIL_IF(UTHCheckPacketMatchResults(p[i], sid, (uint32_t *)&results[i][0], numsigs) == 0);
 
         TimeSetIncrementTime(2);
 
         SCLogDebug("packet %d flag %s", i, p[i]->flags & PKT_HAS_TAG ? "true" : "false");
         /* see if the PKT_HAS_TAG is set on the packet if needed */
-        bool expect;
-        if (i == 0 || i == 5 || i == 6)
-            expect = false;
-        else
-            expect = true;
-        if (((p[i]->flags & PKT_HAS_TAG) ? true : false) != expect)
-            goto cleanup;
+        bool expect = (i == 0 || i == 5 || i == 6);
```

Comment:
this init is the reverse of the original code, but it still passes... wonder why...

Reply:
Could it be because the FAIL_IF macro check was also reversed? Before, the test would fail if expect were not different, and now it fails if not_not different, if that makes sense?